scripts/llnl/common_build_functions.py

```diff
@@ -125,6 +125,40 @@ def log_failure(prefix, msg, timestamp=""):
     json.dump(info,open(pjoin(prefix,"failed.json"),"w"),indent=2)
 
 
+def setup_clingo():
+    """
+    Attempts to install the clingo answer set programming library
+    if it is not already available as a Python module
+    """
+    try:
+        import clingo
+    except ModuleNotFoundError:
+        import pip
+        pip_ver = pip.__version__
```

Comment:
Should this be moved into uberenv?

Reply:
I can open an Uberenv PR, presumably we'd want the concretizer to be configurable via just the JSON file?

Reply:
yes please